perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1005,6 +1010,509 @@ public static boolean isAuthorizedForAttribute(PerunSession sess, ActionType act
+	 * @param policyCollections policy collections containing only required action type collections
+	 * @param associatedObjects map of object types with actual associated objects
+	 * @return true if principal is privileged to operate on attribute (satisfies at least one collection's policies), false otherwise.
+	 */
+	private static boolean resolveAttributeAuthorization(PerunSession sess, List<AttributePolicyCollection> policyCollections, Map<RoleObject, Set<Integer>> associatedObjects) {
+		for (AttributePolicyCollection policyCollection : policyCollections) {
+			boolean collectionSatisfied = true;
+
+			if (policyCollection.getPolicies().isEmpty()) {
+				throw new InternalErrorException("Policy collection with id " + policyCollection.getId() + " contains no policies.");
+			}
+
+			for (AttributePolicy policy : policyCollection.getPolicies()) {
+				if (!resolvePolicyPrivileges(sess, associatedObjects, policy) && !resolveMembershipPrivileges(sess, policy, associatedObjects)) {
+					collectionSatisfied = false;
+					break;
+				}
+			}
+
+			if (collectionSatisfied) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Resolves single attribute policy - checks, if principal has required role in at least one of retrieved associated objects
+	 * @param sess session
+	 * @param associatedObjects map of object types with actual associated objects
+	 * @param policy attribute policy to be checked
+	 * @return true if principal is privileged for attribute, false otherwise
+	 */
+	private static boolean resolvePolicyPrivileges(PerunSession sess, Map<RoleObject, Set<Integer>> associatedObjects, AttributePolicy policy) {
+		AuthzRoles principalRoles = sess.getPerunPrincipal().getRoles();
+
+		if (policy.getObject().equals(RoleObject.None) && principalRoles.hasRole(policy.getRole())) {
+			return true;
+		}
+
+		if (associatedObjects.containsKey(policy.getObject())) {
+			Set<Integer> objectsToCheck = associatedObjects.get(policy.getObject());
+			for (Integer objectId : objectsToCheck) {
+				if (principalRoles.hasRole(policy.getRole(), policy.getObject().name(), objectId)) {
+					return true;
+				}
+			}
+		} else {
+			throw new InternalErrorException("Objects of type " + policy.getObject().name() + " were not retrieved for attribute policy check.");
+		}
+
+		return false;
+	}
+
+	/**
+	 * Resolve membership privileges of single policy for the principal
+	 *
+	 * @param sess from which will be principal fetched
+	 * @param policy policy which may contain MEMBERSHIP role
+	 * @param associatedObjects map of object types with actual associated objects to which will be membership checked
+	 * @return true if roles contains MEMBERSHIP role and the principal satisfies that, false otherwise.
+	 */
+	private static boolean resolveMembershipPrivileges(PerunSession sess, AttributePolicy policy, Map<RoleObject, Set<Integer>> associatedObjects) {
+		if (policy.getRole().equals(Role.MEMBERSHIP)) {
+			if (policy.getObject() == null) {
```

Comment:
I think we can remove this condition, since role MEMBERSHIP should always have an object assigned.

Reply:
Removed